### PR TITLE
Make os.distribution searchable

### PR DESCRIPTION
**Pull request by supervacuus**
Users of the Native SDK also want to search for the Linux distributions their events came from:  https://github.com/getsentry/sentry-native/issues/943

The corresponding PRs to

* develop docs: https://github.com/getsentry/develop/pull/1227
* relay: https://github.com/getsentry/relay/pull/3443
* Native SDK: https://github.com/getsentry/sentry-native/pull/963

@markushi: please have a quick look if that makes any sense 😅 
cc: @kahest

### CODE REVIEW

File: src/sentry/rules/conditions/event_attribute.py [hunk 1]
```diff
@@ -209,6 +211,20 @@ def _get_attribute_values(self, event: GroupEvent, attr: str) -> Sequence[object
                     response = {}
                 return [response.get(path[1])]
 
+        elif path[0] == "os":
+            if path[1] in ("distribution"):
```
Comment (narsaynorath) [inline src/sentry/rules/conditions/event_attribute.py:215]: If someone were to search for `os.distribution` then we'd get an index error here. I think we can extend the pattern above by removing the `elif len(path) != 2` check and structuring it as:
```python
# remove the len(path) != 2 check

# everything that indexes on 2 segments of the path

if len(path) == 2:
    return []  # No other attribute paths of 2 have been mapped

# The new branch for indexing os.distribution.name/version

return [] # Handles all other cases
```

Everything else looks good to me though

Comment (supervacuus) [reply src/sentry/rules/conditions/event_attribute.py:218]: Thanks! I added delimiting path-length branches and a test that would have failed in the previous version with an `IndexError`.

**Comment by supervacuus** [comment]
@narsaynorath, it seems like the failing test is gone now without any change in this PR. Is there anything else I can do to merge this PR?

**Comment by narsaynorath** [comment]
@supervacuus sorry for not getting back to you! I was looking into the failed test but got pulled away. I have to set a label to trigger the backend tests but I can approve after those pass.

Just to confirm, the intention is to make this field searchable in places like Discover and Dashboards? Anywhere else specifically? We'll have to make a frontend PR to surface this field

**Review by narsaynorath — APPROVED** [review]
Changes look good to me. I'll follow up with @kahest on Monday about what else we expect to get done for this r.e. the UI

**Pull request merged**